server/api_test.go

```diff
@@ -842,6 +842,103 @@ func TestSubscribePartitionClosed(t *testing.T) {
 	}
 }
 
+func TestSubscribeStopPosition(t *testing.T) {
+	defer cleanupStorage(t)
+
+	// Use a central NATS server.
+	ns := natsdTest.RunDefaultServer()
+	defer ns.Shutdown()
+
+	// Configure server.
+	s1Config := getTestConfig("a", true, 5050)
+	s1 := runServerWithConfig(t, s1Config)
+	defer s1.Stop()
+
+	getMetadataLeader(t, 10*time.Second, s1)
+
+	client, err := lift.Connect([]string{"localhost:5050"})
+	require.NoError(t, err)
+	defer client.Close()
+
+	stream := "foo"
+	ctx := context.Background()
+	err = client.CreateStream(ctx, "foo", stream)
+	require.NoError(t, err)
+
+	// subscription to empty stream errors
+	err = client.Subscribe(ctx, stream, nil, lift.StopAtLatestReceived())
+	require.Error(t, err)
+	require.Equal(t, lift.ErrStreamExhausted, err)
```

Comment:
Locally I've added support for stop position in go-liftbridge. One change I made was to introduce a new sentinal error: `ErrStreamExhausted` for when the grpc code `ResourceExhausted` is returned. This meant changing the API slightly, as previous a `ErrReadonlyPartition` would be returned for the code `ResourceExhausted`.

Adding the new error might not actually be necessary though?